--- CMakeLists.txt	2025-02-05 04:57:29.000000000 +0300
+++ CMakeLists.txt.mod	2026-06-18 20:54:19.056718462 +0300
@@ -1,16 +1,9 @@
-cmake_minimum_required(VERSION 3.1)
-
-set(CMAKE_C_COMPILER "gcc")
-set(CMAKE_CXX_COMPILER "g++")
+cmake_minimum_required(VERSION 3.5)
 
 project(tinymaix)
 
-set(CMAKE_AR "ar")
-set(CMAKE_RANLIB "ranlib")
-set(CMAKE_STRIP "strip")
-
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3 " CACHE STRING "c flags")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 " CACHE STRING "c++ flags")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} " CACHE STRING "c flags")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} " CACHE STRING "c++ flags")
 aux_source_directory(${CMAKE_CURRENT_SOURCE_DIR}/src lib_tinymaix) 
 include_directories( . ${CMAKE_CURRENT_SOURCE_DIR}/include)
 
