--- include/tm_port.h	2025-02-05 04:57:29.000000000 +0300
+++ include/tm_port.h.mod	2026-06-24 12:14:01.553888966 +0300
@@ -29,7 +29,7 @@
 /******************************* PORT CONFIG  ************************************/
 #define TM_ARCH         TM_ARCH_CPU
 #define TM_OPT_LEVEL    TM_OPT0
-#define TM_MDL_TYPE     TM_MDL_INT8
+#define TM_MDL_TYPE     TM_MDL_FP32
 #define TM_FASTSCALE    (0)         //enable if your chip don't have FPU, may speed up 1/3, but decrease accuracy
 #define TM_LOCAL_MATH   (0)         //use local math func (like exp()) to avoid libm
 #define TM_ENABLE_STAT  (1)         //enable mdl stat functions
