--- configure.in	2013-04-07 03:20:18.000000000 -0700
+++ configure.in	2013-05-13 08:56:53.000000000 -0700
@@ -243,8 +243,11 @@
               elif xcodebuild -version -sdk macosx10.7 Path >/dev/null 2>&1 ; then
                  mac_version_min="-mmacosx-version-min=10.4"
                  mac_sysroot="-isysroot `xcodebuild -version -sdk macosx10.7 Path`"
+              elif xcodebuild -version -sdk macosx10.8 Path >/dev/null 2>&1 ; then
+                 mac_version_min="-mmacosx-version-min=10.4"
+                 mac_sysroot="-isysroot `xcodebuild -version -sdk macosx10.8 Path`"
               else
-                 AC_MSG_ERROR([Couldn't find 10.5, 10.6, or 10.7 SDK])
+                 AC_MSG_ERROR([Couldn't find 10.5, 10.6, 10.7, or  10.8 SDK])
               fi
            esac
 
