--- configure.in.orig	2020-10-02 14:55:01.000000000 -0700
+++ configure.in	2020-10-02 14:53:36.000000000 -0700
@@ -210,7 +210,7 @@
         if test "x$enable_mac_universal" = "xyes" ; then
            case `xcodebuild -version | sed -n 's/Xcode \(.*\)/\1/p'` in
 
-           [12]*|3.0|3.1)
+           3.0|3.1)
               dnl In pre-3.2 versions of Xcode, xcodebuild doesn't
               dnl support -sdk, so we can't use that to look for
               dnl SDKs.  However, in those versions of Xcode, the
@@ -258,8 +258,17 @@
               elif xcodebuild -version -sdk macosx10.12 Path >/dev/null 2>&1 ; then
                  mac_version_min="-mmacosx-version-min=10.4"
                  mac_sysroot="-isysroot `xcodebuild -version -sdk macosx10.12 Path`"
+              elif xcodebuild -version -sdk macosx10.13 Path >/dev/null 2>&1 ; then
+                 mac_version_min="-mmacosx-version-min=10.4"
+                 mac_sysroot="-isysroot `xcodebuild -version -sdk macosx10.13 Path`"
+              elif xcodebuild -version -sdk macosx10.14 Path >/dev/null 2>&1 ; then
+                 mac_version_min="-mmacosx-version-min=10.4"
+                 mac_sysroot="-isysroot `xcodebuild -version -sdk macosx10.14 Path`"
+              elif xcodebuild -version -sdk macosx10.15 Path >/dev/null 2>&1 ; then
+                 mac_version_min="-mmacosx-version-min=10.4"
+                 mac_sysroot="-isysroot `xcodebuild -version -sdk macosx10.15 Path`"
               else
-                 AC_MSG_ERROR([Could not find 10.5 to 10.12 SDK.])
+                 AC_MSG_ERROR([Could not find 10.5 to 10.15 SDK.])
               fi
            esac
 
