--- lib/mosquitto.c
+++ lib/mosquitto.c
@@ -24,6 +24,10 @@ and the Eclipse Distribution License is available at
 #include <strings.h>
 #endif
 
+#if defined(__APPLE__)
+#  include <mach/mach_time.h>
+#endif
+
 #include "logging_mosq.h"
 #include "mosquitto.h"
 #include "mosquitto_internal.h"
--- src/logging.c
+++ src/logging.c
@@ -22,6 +22,9 @@ Contributors:
 #include <syslog.h>
 #endif
 #include <time.h>
+#if defined(__APPLE__)
+#include <sys/time.h>
+#endif
 
 #ifdef WITH_DLT
 #include <sys/stat.h>
