--- sqlite3.c.orig	2019-09-09 15:34:29.906166574 +0200
+++ sqlite3.c	2019-09-09 15:48:34.737157094 +0200
@@ -30733,6 +30733,14 @@
 ** testing and sandboxing.  The following array holds the names and pointers
 ** to all overrideable system calls.
 */
+static int sqlite3_trampoline_ioctl(int d, int request, ...) {
+  va_list ap;
+  int rc;
+  va_start(ap, request);
+  rc = ioctl(d, request, va_arg(ap, void*));
+  va_end(ap);
+  return rc;
+}
 static struct unix_syscall {
   const char *zName;            /* Name of the system call */
   sqlite3_syscall_ptr pCurrent; /* Current value of the system call */
@@ -30883,7 +30891,7 @@
 #endif
 #define osLstat      ((int(*)(const char*,struct stat*))aSyscall[27].pCurrent)
 
-  { "ioctl",         (sqlite3_syscall_ptr)ioctl,          0 },
+  { "ioctl",         (sqlite3_syscall_ptr)sqlite3_trampoline_ioctl,          0 },
 #define osIoctl ((int(*)(int,int,...))aSyscall[28].pCurrent)
 
 }; /* End of the overrideable system calls */
